packages/snaps-controllers/src/interface/utils.ts

```diff
@@ -0,0 +1,93 @@
+import { NodeType, assert } from '@metamask/snaps-sdk';
+import type {
+  Component,
+  Input,
+  FormState,
+  InterfaceState,
+} from '@metamask/snaps-sdk';
+
+/**
+ * Finds the previous value of the input in the old state or sets the input to null.
+ *
+ * @param state - The interface state.
+ * @param component - The Input component.
+ * @returns The input state.
+ */
+export const constructInputState = (
+  state: InterfaceState,
+  component: Input,
+) => {
+  return component.value ?? state[component.name] ?? null;
+};
+
+/**
+ * Finds the previous value of the input in the old state form or sets the input to null.
+ *
+ * @param state - The interface state.
+ * @param component - The Input component.
+ * @param form - The parent form name of the input.
+ * @returns The input state.
+ */
+export const constructFormState = (
+  state: InterfaceState,
+  component: Input,
+  form: string,
+) => {
+  const oldFormState = state[form] as FormState;
+  const oldInputState = oldFormState?.[component.name];
+  return component.value ?? oldInputState ?? null;
```

Comment:
Should we prefer `value` over `oldInputState`? 🤔

Reply:
Yes because the snap developer should be able to override an input on an update.